platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/admin/ImportMigrationConfigurationAdminContext.java

```diff
@@ -0,0 +1,312 @@
+                .collect(Collectors.toMap(ImportMigrationConfigurationAdminEntry::getPid,
+                        Function.identity()));
+        this.exportedFactoryServices = Stream.of(entries)
+                .filter(ImportMigrationConfigurationAdminEntry::isManagedServiceFactory)
+                .collect(Collectors.groupingBy(ImportMigrationConfigurationAdminEntry::getFactoryPid));
+        context.getReport()
+                .doAfterCompletion(this::deleteUnexportedConfigurationsAfterCompletion);
+    }
+
+    private static boolean isManagedServiceFactory(Configuration cfg) {
+        return cfg.getFactoryPid() != null;
+    }
+
+    private static boolean isManagedService(Configuration cfg) {
+        return cfg.getFactoryPid() == null;
+    }
+
+    @Override
+    public Stream<ImportMigrationEntry> entries() {
+        if (!isValid) {
+            return Stream.empty();
+        }
+        return Stream.concat(exportedServices.values()
+                        .stream(),
+                exportedFactoryServices.values()
+                        .stream()
+                        .flatMap(List::stream));
+    }
+
+    @Override
+    public ImportMigrationEntry getEntry(Path path) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    @Override
+    public Stream<ImportMigrationEntry> entries(Path path) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    @Override
+    public Stream<ImportMigrationEntry> entries(Path path, PathMatcher filter) {
+        throw new IllegalStateException("should not be called");
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    Configuration createConfiguration(ImportMigrationConfigurationAdminEntry entry)
+            throws IOException {
+        // Question: should we use the bundle location that was exported???
+        if (entry.isManagedServiceFactory()) {
+            return configurationAdmin.createFactoryConfiguration(entry.getFactoryPid(), null);
+        }
+        return configurationAdmin.getConfiguration(entry.getPid());
+    }
+
+    // PMD.DefaultPackage - designed to be called only from ImportMigrationConfigurationAdminEntry within this package
+    @SuppressWarnings("PMD.DefaultPackage")
+    Configuration getMemoryConfiguration(ImportMigrationConfigurationAdminEntry entry) {
+        final String fpid = entry.getFactoryPid();
+
+        if (fpid != null) {
+            final List<Configuration> mcfgs = memoryFactoryServices.get(fpid);
+
+            if (mcfgs == null) {
+                return null;
+            }
+            // search for it based on the felix file install property
+            final Path epath = getPathFromConfiguration(entry.getProperties(),
+                    () -> String.format("path '%s'", entry.getPath()));
+
+            if (epath != null) {
```

Comment:
✏️ Inverting this test will reduce the complexity of the method.

Reply:
Are you suggesting removing ``if (epath != null) {}`` and changing ``if (epath.equals(mpath)) {`` for ``if (mpath.equals(epath)) {``?

If so then I am not a fan as we would now enter the loop for nothing.